vm/src/obj/objstr.rs

```diff
@@ -425,7 +425,7 @@ impl PyString {
     /// uppercase character and the remaining characters are lowercase.
     #[pymethod]
     fn title(&self, _vm: &VirtualMachine) -> String {
-        let mut title = String::new();
+        let mut title = String::with_capacity(self.value.len());
```

Comment:
Could we also do the same in `__mul__`?

Reply:
Yeah, and in a few other spots as well. I have it ready but I intended to do the other batch in another PR.

Reply:
Ehh I'll just add another commit

Reply:
Actually, it gets a bit messy so I'll do that in another PR